### PR TITLE
Move s[n]printf to lib for release

### PR DESCRIPTION
**Pull request by lifeix**
@anthonyzxu

### CODE REVIEW

**Comment by lifeix** [comment]
Seems OK

File: lib/sprintf.c [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+#endif
+
+#define PRINT_STRING_MAX_LEN 4096
+
+/** Command for the emit function: copy string to output. */
+#define PRINT_CMD_COPY			0x00000000
+
+/** Command for the emit function: fill output with first character. */
+#define PRINT_CMD_FILL			0x00000001
+
```
Comment (JasonChenCJ) [inline lib/sprintf.c]: PRINT_CMD_COPY & PRINT_CMD_FILL is dup with header file

File: debug/printf.c [hunk 2]
```diff
@@ -34,622 +34,6 @@
 #include <hv_arch.h>
```
Comment (JasonChenCJ) [inline debug/printf.c]: why dont we move the whole printf.c into lib directly?

Comment (anthonyzxu) [reply debug/printf.c]: some functions are used only for debug, so put them under debug/ , and reduce the code size.

**Comment by lijinxia** [comment]
verified +1

**Pull request merged**